xqueue.lua

```diff
@@ -547,7 +537,43 @@ function M.upgrade(space,opts,depth)
 	end
 	self.have_runat = have_runat
 
+	---@type table<string, { counts: {}, transition: {} }>
+	local stat_tube = {}
+	if self.fields.tube and type(opts.tube_stats) == 'table' then
+		for _, tube_name in ipairs(opts.tube_stats) do
+			if type(tube_name) == 'string' then
+				stat_tube[tube_name] = {
+					counts = {},
+					transition = {},
+				}
+			end
+		end
+	end
 
+	if not self._stat then
+		self._stat = {
+			counts = {};
+			transition = {};
+			tube = stat_tube;
```

Comment:
this is _stat.tube, it will not be returned in :stats() and won't be collected. the internal table is created to remove som extra if type(...) checks in _on_replace trigger